misc/memory-connection-limits/tests/max_percentage.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Protocol Labs.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+mod util;
+use libp2p_core::Multiaddr;
+use libp2p_identity::PeerId;
+use libp2p_memory_connection_limits::*;
+use sysinfo::{RefreshKind, SystemExt};
+use util::*;
+
+use libp2p_swarm::{dial_opts::DialOpts, DialError, Swarm};
+use libp2p_swarm_test::SwarmExt;
+use rand::{rngs::OsRng, Rng};
+
+#[test]
+fn max_percentage() {
+    let connection_limit = OsRng.gen_range(1..30);
```

Comment:
No randomness please, make it a property-based test instead.

Reply:
This follows the [`max_outgoing`](https://github.com/libp2p/rust-libp2p/blob/f95f39f634d08dad8df72d7ecf92312f9ad2d826/misc/connection-limits/src/lib.rs#L389) test in static connection limits crate.  

And using `quickcheck` won't work here as it generates multiple tests that run in parallel (unless we supply `--test-threads 1` to `cargo test`

How about using a fixed value, say, 20?

Reply:
> This follows the [`max_outgoing`](https://github.com/libp2p/rust-libp2p/blob/f95f39f634d08dad8df72d7ecf92312f9ad2d826/misc/connection-limits/src/lib.rs#L389) test in static connection limits crate.

Touché! I was not aware of that :D

> How about using a fixed value, say, 20?

I'd prefer that :)

Reply:
Fixed